### PR TITLE
[WIP] EDXOLDMNG-62 Do not display Old Mongo courses in Studio

### PR DESCRIPTION
**Pull request by UvgenGen**
[EDXOLDMNG-62](https://youtrack.raccoongang.com/agiles/104-993/105-2666?issue=EDXOLDMNG-62) Do not display Old Mongo courses in Studio

### CODE REVIEW

File: xmodule/modulestore/mixed.py [hunk 1]
```diff
@@ -295,6 +295,11 @@ def get_course_summaries(self, **kwargs):
                     log.warning(
                         "Modulestore %s have duplicate courses %s; skipping from result.", store, course_id
                     )
+                # Skip course if it's from old mongo modulestore.
```
Comment (GlugovGrGlib) [inline xmodule/modulestore/mixed.py:298]: it seems that this changes not required for closing access to studio

File: openedx/core/djangoapps/content/course_overviews/tests/test_tasks.py [hunk 2]
```diff
@@ -12,8 +12,8 @@
 class BatchedAsyncCourseOverviewUpdateTests(ModuleStoreTestCase):  # lint-amnesty, pylint: disable=missing-class-docstring
     def setUp(self):
         super().setUp()
-        self.course_1 = CourseFactory.create(default_store=ModuleStoreEnum.Type.mongo)
-        self.course_2 = CourseFactory.create(default_store=ModuleStoreEnum.Type.mongo)
+        self.course_1 = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
```
Comment (GlugovGrGlib) [inline openedx/core/djangoapps/content/course_overviews/tests/test_tasks.py:15]: This tests changes may be not required

File: lms/djangoapps/courseware/management/commands/tests/test_dump_course.py [hunk 3]
```diff
@@ -107,8 +107,8 @@ def test_dump_course_ids_with_filter(self):
 
     def test_dump_course_ids(self):
         output = self.call_command('dump_course_ids')
-        dumped_courses = output.strip().split('\n')
-        course_ids = {str(course_id) for course_id in self.loaded_courses}
+        dumped_courses = (output.strip() or []) and output.strip().split('\n')
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply lms/djangoapps/courseware/management/commands/tests/test_dump_course.py:110]: Without it if output is empty dumped_courses will be equal `['']`

File: cms/djangoapps/contentstore/tests/test_contentstore.py [hunk 4]
```diff
@@ -1549,6 +1554,10 @@ def test_get_html(handler):
         )
         course_key = course_items[0].id
 
+        # course_handler raise 404 for old mongo course
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply cms/djangoapps/contentstore/tests/test_contentstore.py:1559]: updated

File: cms/djangoapps/contentstore/tests/test_course_listing.py [hunk 5]
```diff
@@ -164,7 +164,6 @@ def test_courses_list_with_ccx_courses(self):
 
     @ddt.data(
         (ModuleStoreEnum.Type.split, 2),
-        (ModuleStoreEnum.Type.mongo, 1)
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply cms/djangoapps/contentstore/tests/test_course_listing.py]: updated

File: cms/djangoapps/contentstore/views/tests/test_course_index.py [hunk 6]
```diff
@@ -100,6 +100,8 @@ def test_is_staff_access(self):
         """
         Test that people with is_staff see the courses and can navigate into them
         """
+        if self.course.id.deprecated:
+            raise SkipTest("Skip test for old mongo course")
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply cms/djangoapps/contentstore/views/tests/test_course_index.py]: updated

File: cms/djangoapps/contentstore/views/tests/test_course_index.py [hunk 7]
```diff
@@ -131,6 +136,9 @@ def test_course_staff_access(self):
         self.check_courses_on_index(course_staff_client)
 
     def test_json_responses(self):
+        if self.course.id.deprecated:
+            raise SkipTest("Skip test for old mongo course")
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply cms/djangoapps/contentstore/views/tests/test_course_index.py]: updated

File: openedx/core/djangoapps/content/course_overviews/models.py [hunk 8]
```diff
@@ -673,6 +673,10 @@ def get_all_courses(cls, orgs=None, filter_=None):
         if filter_:
             course_overviews = course_overviews.filter(**filter_)
 
+        # Exclude old mongo courses
+        deprecated_course_ids = [id for id in course_overviews.values_list('id', flat=True) if id.deprecated]
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply openedx/core/djangoapps/content/course_overviews/models.py]: updated

File: lms/djangoapps/course_api/tests/test_views.py [hunk 9]
```diff
@@ -419,7 +419,7 @@ def test_too_many_courses(self):
         self.setup_user(self.audit_user)
 
         # These query counts were found empirically
-        query_counts = [50, 46, 46, 46, 46, 46, 46, 46, 46, 46, 16]
+        query_counts = [51, 47, 47, 47, 47, 47, 47, 47, 47, 47, 17]
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply lms/djangoapps/course_api/tests/test_views.py:422]: in `openedx/core/djangoapps/content/course_overviews/models.py get_all_courses` we add excluding old mongo courses

**Comment by GlugovGrGlib** [comment]
The PR was opened to upstream https://github.com/openedx/edx-platform/pull/30933

**Pull request closed without merging**